src/NUnitTestExtractor/NUnitTestExtractor/Program.cs

```diff
@@ -32,68 +33,132 @@ static void Main(string[] args)
                         {
                             Console.Error.WriteLine(e.GetType() + ": Please enter either Namespace, Class, or Function for level");
                             Environment.Exit(-1);
                         }
+                        finally
+                        {
+                            if (writer != null)
+                            {
+                                writer.Dispose();
+                                
+                            }
+                        }
+
                     }
                 }
             }
         }
+
         /// <summary>
-        /// While reflection is not implemented method writes each dll verified to be existing to either a txt file or stdout along with the specified level
+        /// Writes each DLL file path along with either a namespace, class or function which contains tests(dependent on level specified by user)
         /// </summary>
-        /// <param name="name"> the name of the txt file that data is being appended to </param>
-        /// <param name="outputDirectory"> the directory in that the file is being saved to </param>
-        /// <param name="dllDirectories"> list of all dll directories which the user wishes to append to a file or stdout </param>
-        /// <param name="level"> Specifies the level of granuality to use for the output tests entered by user. </param>
-        private static void WriteTestDllAndName(string name, string outputDirectory, IList<string> dllDirectories, Level level)
+        /// <param name="writer"> StreamWriter used to write the dll path and data, either to a txt file or stdout </param>
+        /// <param name="dll"> The dll path which is written </param>
+        /// <param name="data"> The test information corrosponding to the dll file </param>
+        private static void WriteTestDllAndName(StreamWriter writer, string dll, string data)
         {
-            //user entered some dlls
-            if (dllDirectories.Count != 0)
-            {
-                StreamWriter writer;
+            writer.Write(dll + "|" + data + Environment.NewLine);
+        }
 
-                if (outputDirectory == null)
+        /// <summary>
+        /// Uses reflection in order to search user submitted dlls to find either namespaces, classes, or functions which contain tests, then uses WriteTestDllAndName() to
+        /// write data to either stdout or a txt file
+        /// </summary>
+        /// <param name="dlls">The dlls which are submitted by user</param>
+        /// <param name="level">Specifies the level of granuality to use for the output: either namepspace, class or function</param>
+        /// <param name="writer">The StreamWriter which will be passed to WriteTestDllAndName() </param>
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Reliability", "CA2001:AvoidCallingProblematicMethods", MessageId = "System.Reflection.Assembly.LoadFrom")]
```

Comment:
MSDN says LoadFrom is a "problematic function".  I'm not sure exactly what kind of problems they're referring to.  They reference this page:
https://blogs.msdn.microsoft.com/suzcook/2003/09/19/loadfile-vs-loadfrom/
which talks about using this instead:
https://msdn.microsoft.com/en-us/library/system.reflection.assembly.reflectiononlyloadfrom(v=vs.110).aspx
but that doesn't automatically load depndencies and then tells you to handle this event to load the dependencies:
https://msdn.microsoft.com/en-us/library/system.appdomain.reflectiononlyassemblyresolve(v=vs.110).aspx

For this pull request, using Assembly.LoadFrom() should be OK, but maybe afterwards you can investigate using some of the solutions they suggest in those links.